sdk/data/azcosmos/example_test.go

```diff
@@ -0,0 +1,201 @@
+			Paths: []string{"/myPartitionKey"},
+		},
+		ETag: "someEtag",
+		IndexingPolicy: &azcosmos.IndexingPolicy{
+			IncludedPaths: []azcosmos.IncludedPath{},
+			ExcludedPaths: []azcosmos.ExcludedPath{},
+			Automatic:     false,
+			IndexingMode:  azcosmos.IndexingModeNone,
+		},
+	}
+
+	// Replace container properties
+	resp, err = container.Replace(ctx, updatedProperties, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Read the manual throughput property
+	throughputResponse, _ := container.ReadThroughput(ctx, nil)
+	_, err = throughputResponse.ThroughputProperties.ManualThroughput()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Replace manual throughput property
+	newScale := azcosmos.NewManualThroughputProperties(500)
+	_, err = container.ReplaceThroughput(ctx, *newScale, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Migrate from manual throughput to autoscale
+	_, err = throughputResponse.ThroughputProperties.AutoscaleMaxThroughput()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	newScale = azcosmos.NewAutoscaleThroughputProperties(10000)
+	_, err = container.ReplaceThroughput(ctx, *newScale, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// ===== 4. Item CRUD =====
+
+	// Items in an Azure Cosmos container are uniquely identified by their id and partition key value.
+	pk, err := azcosmos.NewPartitionKey("newPartitionKey")
+
+	item := map[string]string{
+		"id":             "1",
+		"value":          "2",
+		"myPartitionKey": "newPartitionKey",
+	}
+
+	// Create item.
+	itemResponse, err := container.CreateItem(ctx, pk, item, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Read item.
+	itemResponse, err = container.ReadItem(ctx, pk, "1", nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Item response body
+	var itemResponseBody map[string]interface{}
+	err = json.Unmarshal(itemResponse.Value, &itemResponseBody)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// ===== 5. Session consistency =====
+
+	itemResponse, err = container.UpsertItem(ctx, pk, item, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	itemSessionToken := itemResponse.SessionToken()
+	itemResponse, err = container.ReadItem(ctx, pk, "1", &azcosmos.CosmosItemRequestOptions{SessionToken: itemSessionToken})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// ===== 6. Optimistic Concurrency Etag PreConditionFail =====
+
+	// Azure Cosmos DB supports optimistic concurrency control to prevent lost updates or deletes and detection of conflicting operations.
+	numberRetry := 3
+	item = map[string]string{
+		"id":             "1",
+		"value":          "newValue",
+		"myPartitionKey": "newPartitionKey",
+	}
+
+	// Check the item response status code. If an error is imitted and the response code is 412 then retry operation.
+	err = retryOptimisticConcurrencyExample(numberRetry, 1000*time.Millisecond, func() (int, error) {
+		itemResponse, err = container.ReadItem(ctx, pk, "1", nil)
+		return itemResponse.RawResponse.StatusCode, err
```

Comment:
Any failure (like 412) is surfaced in the `err`, not in the itemResponse.

Basically you need to check if there is any err and if so, attempt to extract the statusCode to see if it's 412:

```
var httpErr azcore.HTTPResponse
if errors.As(err, &httpErr) && httpErr.RawResponse().StatusCode == 412 {
    // needs to be retried
}
```